src/core/util.ts

```diff
@@ -59,6 +59,23 @@ export function makeBacktrace(stack: string, shift = 0): BacktraceFrame[] {
   }
 }
 
+export function getCauses(notice: Partial<Notice>) {
+  if (notice.cause) {
+    const causes =[]
+    let cause = notice as Error
+    while (causes.length < 3 && (cause = cause.cause) as Error) {
```

Comment:
Nice, I haven't used `as` in this way. I had to try it out in a playground!
<img width="634" alt="image" src="https://user-images.githubusercontent.com/5813382/183753407-a65d4bef-140d-492d-81cb-f9ba2342d2db.png">
But I'm curious to see how `(cause = cause.cause) as Error` compiles into browser parseable expression 😅